docs/dotnet/dotnet-signalr-metrics.md

```diff
@@ -0,0 +1,68 @@
+# Semantic Conventions for SignalR metrics emitted by ASP.NET Core
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for SignalR metrics emitted by .NET components and runtime.
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Metric: `signalr_http_transport.connection.duration`](#metric-signalr_http_transportconnectionduration)
+- [Metric: `signalr_http_transport.active_connections`](#metric-signalr_http_transportactive_connections)
+
+<!-- tocstop -->
+
+## Metric: `signalr_http_transport.connection.duration`
+
+**Status**: [Experimental][DocumentStatus]
+
+This metric is required.
+
+<!-- semconv metric.signalr.server.connection.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `signalr_http_transport.connection.duration` | Histogram | `s` | The duration of connections on the server. [1] |
+
+**[1]:** **TODO: why http_transport? can we report protocol name as dimension and not a metric name?**
+**TODO: we should do signalr.server**
+Corresponding `EventCounter` name is `connections-duration`; Meter name is `Microsoft.AspNetCore.Http.Connections`
+<!-- endsemconv -->
+
+<!-- semconv metric.signalr.server.connection.duration(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `signalr.http_transport.status` | string | TODO [1] | `TODO` | Recommended |
+| `signalr.http_transport.transport` | string | TODO [2] | `TODO` | Recommended |
+
+**[1]:** **TODO: why not signalr.status? What's HTTP-specific about it**
+
+**[2]:** **TODO: why not network.transport?**
+<!-- endsemconv -->
+
+## Metric: `signalr_http_transport.active_connections`
```

Comment:
Porting active comment threads from the word doc:

## Trask Stalnaker
I think existing otel sem conv conventions probably lean towards this being signalr_http_transport.connections.count (or possibly just signalr_http_transport.connections, see https://github.com/open-telemetry/semantic-conventions/issues/50)see https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#use-count-instead-of-pluralization


Otel does have db.client.connections.usage, but I think(?) this should be db.client.connections.count, and I just opened https://github.com/open-telemetry/semantic-conventions/issues/201


Comparing this to otel's http.server.active_requests is also reasonable, but I think the above modeling is probably going to be more consistent long-term.

July 21, 2023 at 8:36 AM
## Trask Stalnaker
I'm having second thoughts about this one already 🤔

July 21, 2023 at 8:45 AM
## Trask Stalnaker
Opened https://github.com/open-telemetry/semantic-conventions/issues/202

July 21, 2023 at 9:18 AM
## Noah Falk
Yeah, I modeled it originally on http.server.active_requests, but I'd be just fine with current_connections.  connections.count and connections are a little more ambiguous because there are two different counts that might refer to, the count of all connections ever made or the count of connections currently open.

July 21, 2023 at 5:36 PM